### DISCUSSION TITLE
Very high CPU usage (%70) when running YOLOV8m model on  i7-1255U 1.70GHz 16GB Ram computer

### DISCUSSION THREAD

**Discussion by poycrypto** [ultralytics/ultralytics#15063] [Q&A]
Hello. I developed a simple Python program with openCV and Ultralytics which uses a custom model to detect two different objects in real time via USB camera. When I run the program, CPU usage is very high. You can see my code and task manager screenshot below. I am suspicious that it may be due to some part of my code. I would appreciate if you could have a look. Thanks in advance.
![Ekran görüntüsü 2024-08-09 152951](https://github.com/user-attachments/assets/a69c1892-1b92-4705-ba14-8db95af132c9)

```
import cv2, math
from roboflow import Roboflow
from ultralytics import YOLO

def main():

    cap = cv2.VideoCapture(1, cv2.CAP_DSHOW)
    cap.set(cv2.CAP_PROP_FRAME_WIDTH, 1280)
    cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 720)
    cap.set(cv2.CAP_PROP_FOURCC, cv2.VideoWriter_fourcc(*"MJPG"))

    model = YOLO('handle_cover_m2.pt')

    while True:
                
        success, frame = cap.read()

        results = model(frame, stream=True, iou = 0.1)
        for r in results:
            boxes = r.boxes
            
            for box in boxes:

                # bounding box
                confidence = math.ceil(box.conf[0]*100)/100
                if confidence >=0.6:
                    x1, y1, x2, y2 = box.xyxy[0]
                    x1, y1, x2, y2 = int(x1), int(y1), int(x2), int(y2) # convert to int values
                    #put box in cam
                    cv2.rectangle(frame, (x1, y1), (x2, y2), (0, 255, 0), 1)

                    #put text in cam
                    org = [x1, y1-5]
                    org2 = [x2, y2-5]
                    font = cv2.FONT_HERSHEY_SIMPLEX
                    fontScale = 0.5
                    color = (0, 255, 0)
                    thickness = 1
                    cv2.putText(frame, ("%.2f") % confidence, org, font, fontScale, color, thickness)
                else:
                    pass       
            
        cv2.imshow("handle_cover_detection", frame)
        if cv2.waitKey(1) == ord('q'):
            break

    cap.release()
    cv2.destroyAllWindows()

if __name__ == "__main__":
    
    main()
```

**Comment by pderrenger** [comment]
@poycrypto hello,

High CPU usage can be common when running real-time object detection, especially with models like YOLOv8. To help mitigate this, ensure you are using the latest versions of Ultralytics and OpenCV. Additionally, consider optimizing your code by reducing the frame rate or using a more efficient model. You might also want to offload some processing to the GPU if available. If the issue persists, please check if the problem is reproducible with the latest package versions and let us know.

**Comment by poycrypto** [comment]
Thanks a lot for the reply. I integrated openvino and now inference time went from 500ms to 50 ms which is amazing. I observe that iGPU is enabled and usage increases to %40. But CPU usage is still %70. Is there a way to offload some of CPU process to iGPU ?

Note: I only have Intel Integrated Graphics card.

**Reply by glenn-jocher** [reply]
@poycrypto i'm glad to hear about the improvement with OpenVINO! To further offload processing to the iGPU, ensure you're using the latest Intel drivers and OpenVINO toolkit. You might also consider optimizing your model further or adjusting the inference precision. For more details, check the OpenVINO documentation.
